pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -235,7 +237,10 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
       SegmentMetadata segmentMetadata = getSegmentMetadata(tempDecryptedFile, tempSegmentDir, metadataProviderClass);
 
       // Fetch segment name
-      String segmentName = segmentMetadata.getName();
+      if (!StringUtils.isEmpty(segmentName)) {
```

Comment:
Don't you need to replace the segment name in the metadata file and rewrite the tar file?

Reply:
This is just to override the segment name when registering the segment in a table. So no need to rewrite the metadata file.

Reply:
Does segment load work fine? Used to be that the name in the segment metadata is the source of truth for segment name. Has that changed?